pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -732,6 +736,28 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, _ v1.PodStatus, podStat
 			continue
 		}
 
+		// limit pod restart count if set
+		restartCountLimitStr, ok := pod.Annotations[podRestartCountLimitAnn]
+		if ok {
+			if len(restartCountLimitStr) > 0 {
+				restartCountLimit, err := strconv.Atoi(restartCountLimitStr)
+				if err != nil {
+					glog.V(5).Infof("Failed to parse restart count limit in pod %v: %+v", format.Pod(pod), err)
+				} else {
+					currentRestartCount := 0
+					for _, containerStatus := range podStatus.ContainerStatuses {
+						currentRestartCount += containerStatus.RestartCount
+					}
+					if currentRestartCount > restartCountLimit {
+						msg := fmt.Sprintf("Stop restarting container %+v in pod %v due to exceeding pod restart count limit", container, format.Pod(pod))
+						startContainerResult.Fail(errors.New("ExceededRestartCountLimit"), msg)
```

Comment:
What's the status of Pod when `restartCountLimit` exceeded? Do we need to make it as `Failed`?